java/src/test/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImplTest.java

```diff
@@ -0,0 +1,132 @@
+package com.xxAMIDOxx.xxSTACKSxx.api.v1.menu.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.xxAMIDOxx.xxSTACKSxx.api.v1.menu.SearchMenuResult;
+import com.xxAMIDOxx.xxSTACKSxx.api.v1.menu.SearchMenuResultItem;
+import com.xxAMIDOxx.xxSTACKSxx.model.Menu;
+import com.xxAMIDOxx.xxSTACKSxx.service.MenuService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.xxAMIDOxx.xxSTACKSxx.api.v1.menu.utilities.HelperMethods.getBaseURL;
+import static com.xxAMIDOxx.xxSTACKSxx.api.v1.menu.utilities.HelperMethods.toJson;
+import static java.util.UUID.randomUUID;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class MenuControllerImplTest {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @MockBean
+    private MenuService service;
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @Test
+    public void whenCalledForMenuReturnsOK() {
+        var entity = this.testRestTemplate.getForEntity(
+                getBaseURL(port) + "/v1/menu/", Map.class);
```

Comment:
Map.class or SearchMenuResult.class ?

Reply:
Also, should be without a trailing slash according to the swagger. Can you check?